### PR TITLE
[sinks] Move ComputeSink structs out of storage

### LINKED ISSUE #14202: Remove ComputeSink* structs etc from storaged package
Follow-up on #14009.  Should just be code movement

### PR DESCRIPTION
**Pull request by cjubb39**
Move ComputeSinkDesc and ComputeSinkConnection out of storage and into `compute-client`.  Given the current code structure, it needs to be `compute-client` because `compute` depends on `compute-client`.

We change some proto references but the definitions of the messages are unchanged so there should be no backwards compat issue

### Motivation
Fixes #14202

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-protobuf` label.
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  -

### CODE REVIEW

**Comment by cjubb39** [comment]
Tagging in @teskje who worked on compute sinks a little bit back in case there's a suggestion for a better place to put stuff in the compute layer

File: src/compute-client/src/sinks.proto [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import "google/protobuf/empty.proto";
+
+import "repr/src/antichain.proto";
+import "repr/src/global_id.proto";
+import "repr/src/relation_and_scalar.proto";
+import "storage/src/controller.proto";
+
+package mz_compute_client.sinks;
+
+message ProtoComputeSinkDesc {
```
Comment (bkirwi) [inline src/compute-client/src/sinks.proto:20]: Can we remove these from the storage proto, or are they still referenced somewhere?

**Review by bkirwi — COMMENTED** [review]
Question aside, this looks good! Though I'm not the best judge of where code ought to live yet, so I'll let one of the other reviewers check that out too.

File: src/compute-client/src/sinks.proto (hunk 1 above)
Comment (cjubb39) [reply src/compute-client/src/sinks.proto:20]: oof -- good catch, thanks!

**Review by teskje — APPROVED** [review]
LGTM!

**Pull request merged**